File: swift/Sources/ShapesBaseModels/Square.swift
import Foundation
import ShapesBaseCommons

public final class Square: Dimensional {
    private let side: Double

    public init(_ side: Double) {
        let dimOne: Double = side > 0 ? side : 1
        self.side = dimOne
    }

    public func area() -> Double {
        return pow(side, 2)
    }

    public func perimeter() -> Double {
        return 4 * side
    }
}

File: csharp/src-temp/Shapes/Base/Models/Square.cs
using Shapes.Base.Commons;

namespace Shapes.Base.Models;

public sealed class Square : IDimensional {
    private readonly double _side;

    public Square(double side) {
        double dimOne = side > 0 ? side : 1;
        _side = dimOne;
    }

    public double Area() {
        return Math.Pow(_side, 2);
    }

    public double Perimeter() {
        return 4 * _side;
    }
}
